terraform/modules/provision-k8s/k8s-autoscaler.tf

```diff
@@ -0,0 +1,42 @@
+locals {
+  karpenter_provisioner = <<-TEMPLATE
+    apiVersion: karpenter.sh/v1alpha5
+    kind: Provisioner
+    metadata:
+      name: default
+    spec:
+      requirements:
+        - key: karpenter.sh/capacity-type
+          operator: In
+          values: ["on-demand"]
+      limits:
+        resources:
+          cpu: 1000
+      provider:
+        launchTemplate: <launch-template-name>
+        subnetSelector:
+          karpenter.sh/discovery: <cluster-name>
+      ttlSecondsAfterEmpty: 30
+  TEMPLATE
+}
+
+resource "null_resource" "autoscaler-provisioner" {
```

Comment:
You can replace the use of the null_resource and all the monkeying about with the kubeconfig by using [the new `kubernetes_manifest` Terraform resource](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/manifest).

Essentially you can do 
```terraform
resource "kubernetes_manifest" "karpenter-provisioner" {
  manifest = yamldecode(local.karpenter_provisioner)
}
```
That would clean this up a lot!

Reply:
...and remove the need for more mapping outputs to inputs between the modules...